tools/check_pod_errors

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+<< 'heredoc_pod_error_rule'
+The script checks that no POD error are present in any changed file.
+heredoc_pod_error_rule
+
+libs_files=$(git ls-files lib/ | grep '.pm$' | xargs echo)
+tmpfile=$(mktemp)
+tmperrorfile=$(mktemp)
+
+success=1
+if test -n "$libs_files"; then
+    for libfile in $libs_files; do
+        perldoc -T -D "${libfile}" 2>/dev/null 1>"$tmpfile"
+        grep -q "POD ERRORS" "$tmpfile" || continue
+
+        success=0
+        # search if the perdoc output is reporting any error
+        error_line=$(grep -n "POD ERRORS" "$tmpfile" | sed -n '$s/:.*//p')
+        # extract the error message from perldoc output
+        sed -n "$error_line,\$"p "$tmpfile" > "$tmperrorfile"
+        if [[ -n "${GITHUB_ACTIONS}" ]]; then
+            echo "::error file=${libfile}::perldoc ERRORS"
+            # add a notification for each error in the file
+            awk '/Around line/ {          # search for lines containing "Around line"
+                   match($0, /[0-9]+/);   # extract the line number of the error
+                   getline nextline;      # read the error description from the next line
+                   gsub(/^[ \t]+/, "", nextline);  # remove leading whitespace from the error message
+                   print "::error file='"${libfile}"',line="substr($0, RSTART, RLENGTH)"::"nextline;  # print in a github action compliant way
+                }' "$tmperrorfile"
```

Comment:
I wonder if `podchecker` would have been a better choice?

```
*** WARNING: Verbatim paragraph in NAME section at line 15 in file ../os-autoinst-distri-opensuse/lib/qesapdeployment.pm
*** ERROR: You forgot a '=back' before '=head3' at line 447 in file ../os-autoinst-distri-opensuse/lib/qesapdeployment.pm
*** ERROR: You forgot a '=back' before '=head3' at line 467 in file ../os-autoinst-distri-opensuse/lib/qesapdeployment.pm
*** ERROR: You forgot a '=back' before '=head3' at line 995 in file ../os-autoinst-distri-opensuse/lib/qesapdeployment.pm
*** ERROR: '=item' outside of any '=over' at line 1072 in file ../os-autoinst-distri-opensuse/lib/qesapdeployment.pm
../os-autoinst-distri-opensuse/lib/qesapdeployment.pm has 4 pod syntax errors.
*** WARNING: empty section in previous paragraph at line 91 in file ../os-autoinst-distri-opensuse/lib/version_utils.pm
*** ERROR: '=item' outside of any '=over' at line 708 in file ../os-autoinst-distri-opensuse/lib/version_utils.pm
*** ERROR: You forgot a '=back' before '=head2' at line 739 in file ../os-autoinst-distri-opensuse/lib/version_utils.pm
../os-autoinst-distri-opensuse/lib/version_utils.pm has 2 pod syntax errors.
```

Created https://progress.opensuse.org/issues/138521 but if you can give it a try @mpagot? but for now, as is, I don't see reasons to block it

Reply:
I'm not sure if perldoc and perlckeck checks overlap but at least perlcheck can return more info:
- errors (that are probably the same reported by perldoc and that this PR is addressing
- file has not documentation: does not contain any pod commands.
- warnings
- no errors: pod syntax ok

Also perlcheck output needs some adaptation to be compliant with github action annotations

Reply:
> I'm not sure if perldoc and perlckeck checks overlap but at least perlcheck can return more info:
> 
>     * errors (that are probably the same reported by perldoc and that this PR is addressing
> 
>     * file has not documentation: does not contain any pod commands.
> 
>     * warnings
> 
>     * no errors: pod syntax ok
> 
> 
> Also perlcheck output needs some adaptation to be compliant with github action annotations

Indeed, ultimately that will be poo#[138521](https://progress.opensuse.org/issues/138521)